activeBranch/state.sourcing.js

```diff
@@ -0,0 +1,68 @@
+const State = require("state.base");
+const { StateTypes } = require("state.base");
+
+class SourcingState extends State{
+    
+    constructor(creep)
+    {
+        this.creep = creep;
+        this.command = this.creep.memory.command;
+        this.initState = this.creep.memory.state;
+        if(this.command.commandType == "harvest")//If this is a post harvest and the source is predetermined...
```

Comment:
Can be rewritten: if source ID in memory, act with that, and then during action, if it's a post harvest, do not change states.